astropy/coordinates/sky_coordinate.py

```diff
@@ -227,7 +230,7 @@ def _parse_inputs(self, args, kwargs):
                 # coord_kwargs will contain keys like 'ra', 'dec', 'distance'
                 # along with any frame attributes like equinox or obstime which
                 # were explicitly specified in the coordinate object (i.e. non-default).
-                coord_kwargs = _parse_coordinate_arg(args[0], frame, units)
+                coord_kwargs = _parse_coordinate_arg(args[0], frame, units, kwargs)
```

Comment:
but the ``coord_kwargs`` have *not* yet been checked for conflicts - that's what the process of copying from ``coord_kwargs`` to ``valid_kwargs`` below.  If I just add something like
```
valid_kwargs.update(coord_kwargs)
coord_kwargs = {}
```
below this, some of the tests like
```
sc = SkyCoord(1, 2, frame='icrs', unit='deg', equinox='J1999', obstime='J2001')
with pytest.raises(ValueError) as err:
    SkyCoord(sc, equinox='J1999', obstime='J2002')
```
now fail, because the second `SkyCoord` is created just fine (which we *don't* want)